tests/PHPStan/Rules/Comparison/ImpossibleCheckTypeMethodCallRuleTest.php

```diff
@@ -72,6 +72,14 @@ public function testRule(): void
 				'Call to method ImpossibleMethodCall\Foo::isSame() with stdClass and stdClass will always evaluate to true.',
 				78,
 			],
+			[
+				'Call to method ImpossibleMethodCall\Foo::isNotSame() with stdClass and stdClass will always evaluate to false.',
+				81,
+			],
+			[
+				'Call to method ImpossibleMethodCall\Foo::isSame() with *NEVER* and stdClass will always evaluate to false.',
```

Comment:
I don't understand that yet

Reply:
Built another test and apparently the Identical adaptions fix the following
```php
public function foo(\stdClass $a, \stdClass $b): void
	{
		if ($a === $a) {
			assertType('stdClass', $a);
		} else {
			assertType('*NEVER*', $a); // was stdClass before
		}

		if ($b !== $b) {
			assertType('*NEVER*', $b); // was stdClass before
		} else {
			assertType('stdClass', $b);
		}
	}
```

Reply:
I wouldn't expect that at all. I worry this will break `if ($a === $b)` situations.

Reply:
Please add/fix these assertions in NodeScopeResolverTest.

Reply:
added them, I hope I did not understand you wrong. did I?